backend/capellacollab/projects/toolmodels/modelsources/git/github/interface.py

```diff
@@ -0,0 +1,197 @@
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> dict:
+    return json.loads(
+        get_artifact_from_job(
+            project_id,
+            job_id,
+            trusted_path_to_artifact,
+            git_model,
+            git_instance,
+        )
+    )
+
+
+def get_artifact_from_job(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> str:
+    response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/actions/runs/{job_id}/artifacts",
+        headers={
+            "Authorization": f"token {git_model.password}",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "Accept": "application/vnd.github+json",
+        },
+        timeout=2,
+    )
+    response.raise_for_status()
+    artifact_id = response.json()["artifacts"][0]["id"]
+
+    if response.json()["artifacts"][0]["expired"] == "true":
+        raise fastapi.HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={
+                "err_code": "ARTIFACT_EXPIRED",
+                "reason": (
+                    "The latest artifact you are requesting expired. Please rerun your pipline and contact your administrator."
+                ),
+            },
+        )
+    artifact_response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/actions/artifacts/{artifact_id}/zip",
+        headers={
+            "Authorization": f"token {git_model.password}",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "Accept": "application/vnd.github+json",
+        },
+        timeout=2,
+    )
+    artifact_response.raise_for_status()
+
+    return get_file_content(artifact_response, trusted_path_to_artifact)
+
+
+def get_file_content(
+    response: requests.Response, trusted_file_path: str
+) -> str:
+    with zipfile.ZipFile(io.BytesIO(response.content)) as zip_ref:
+        temp_dir = tempfile.mkdtemp()
+        zip_ref.extractall(temp_dir)
+        with open(
+            temp_dir + "/" + trusted_file_path.split("/")[-1],
+            encoding="utf-8",
+        ) as file:
+            content = file.read()
+        shutil.rmtree(temp_dir)
+        return content
+
+
+def get_file_from_repository(
+    trusted_file_path: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> bytes:
+    project_id = get_project_id_by_git_url(git_model, git_instance)
+    public_response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/contents/{parse.quote(trusted_file_path, safe='')}?ref={parse.quote(git_model.revision, safe='')}",
+        timeout=2,
+    )
+    if public_response.ok:
+        response = public_response
+    else:
```

Comment:
I don't like the approach that we have to send two requests. If there is a password set, we should send the authenticated request first. The chance is higher to get a successful response.